lib/analyze-commits.js

```diff
@@ -0,0 +1,72 @@
+const { join } = require("path");
+const fetch = require("node-fetch");
+const eik = require("@eik/cli");
+const common = require("@eik/common");
+
+const messages = {
+  IVERSIONUPDATE: "Version: ✅ Updated Eik version from v%s to v%s.",
+  IVERSIONNOTNEEDED:
+    "Version: ✅ Not needed as the current version of this package already contains these files.",
+  IPUBLISHNOTNEEDED:
+    "Publish: ✅ Not needed as the current version of this package is already published.",
+  IPUBLISHNEEDED:
+    "Version: ✅ Not needed as the current Eik version not yet published.",
+};
+
+/**
+ * Semantic Release "Analyze Commits" hook
+ * Responsible for determining the type of the next release (major, minor or patch). If multiple plugins with a analyzeCommits step are defined, the release type will be the highest one among plugins output.
+ * See https://github.com/semantic-release/semantic-release/blob/master/docs/usage/plugins.md#plugins
+ *
+ * @param {object} options
+ * @param {object} context
+ * @param {object} state
+ */
+module.exports = async function analyzeCommits(options, context, state) {
+  const { version: currentVersion, server, type, name } = state.eikJSON;
+
+  // attempt to version Eik assets
+  try {
+    state.versionToPublish = await eik.version({
+      name,
+      version: currentVersion,
+      type,
+      files: state.eikJSON.files,
+      server,
+      cwd: context.cwd,
+      level: options.level,
+      map: state.eikJSON.map,
+      out: state.eikJSON.out,
+    });
+    context.logger.log(
+      messages.IVERSIONUPDATE,
+      currentVersion,
+      state.versionToPublish
+    );
+    state.publishNeeded = true;
+    return options.level || "patch";
+  } catch (err) {
```

Comment:
Perhaps good to check this for an expected error rather than catching all of them?